eth2near/eth2near-block-relay-rs/src/eth2near_relay.rs

```diff
@@ -276,44 +217,70 @@ impl Eth2NearRelay {
             info!(target: "relay", "== New relay loop ==");
             sleep(Duration::from_secs(12));
 
-            let max_slot_for_submission: u64 = skip_fail!(
-                self.get_max_slot_for_submission(),
-                "Fail to get last slot on Eth",
+            let client_mode: ClientMode = skip_fail!(self.eth_client_contract.get_client_mode(),
+                "Fail to get client mode",
                 self.sleep_time_on_sync_secs
             );
 
-            let mut last_eth2_slot_on_near: u64 = skip_fail!(
-                self.get_last_eth2_slot_on_near(max_slot_for_submission),
-                "Fail to get last slot on NEAR",
-                self.sleep_time_on_sync_secs
-            );
+            let were_submission_on_iter = match client_mode {
+                ClientMode::SubmitLightClientUpdate => self.submit_light_client_update(),
+                ClientMode::SubmitHeader => self.submit_headers()
+            };
+
+            if !were_submission_on_iter {
+                info!(target: "relay", "Sync with ETH network. Sleep {} secs", self.sleep_time_on_sync_secs);
+                thread::sleep(Duration::from_secs(self.sleep_time_on_sync_secs));
+            }
+        }
+    }
+
+    fn submit_light_client_update(&mut self) -> bool {
+        info!(target: "relay", "Submit Light Client Update mode");
+        self.send_light_client_updates_with_checks()
+    }
+
+    fn get_max_block_number(&mut self) -> Result<u64, Box<dyn Error>> {
+        Ok(self.eth_client_contract.get_unfinalized_tail_block_number()?.unwrap_or(
+            self.beacon_rpc_client.get_block_number_for_slot(
+                Slot::new(self.eth_client_contract.get_finalized_beacon_block_slot()?))? + 1) - 1)
+
+    }
+
+    fn submit_headers(&mut self) -> bool {
+        info!(target: "relay", "Submit Headers mode");
+
+        let min_block_number = return_val_on_fail!(self.eth_client_contract.get_last_block_number(),
+                                               "Fail to get last block number",
+                                               false);
```

Comment:
```suggestion
        let min_block_number = return_val_on_fail!(self.eth_client_contract.get_last_block_number(),
                                               "Fail to get last block number",
                                               false) + 1;
```
I think it is better to add `+1` here to avoid confusion about `include/exclude` value in the range for the `get_execution_blocks_between` call.

Reply:
fixed: https://github.com/aurora-is-near/rainbow-bridge/pull/883/commits/230ccb9835d31805895490cf6744d24ec47dba11